PWGHF/D2H/Tasks/taskCharmResoReduced.cxx

```diff
@@ -0,0 +1,264 @@
+
+using namespace o2;
+using namespace o2::soa;
+using namespace o2::analysis;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+enum DecayChannel : uint8_t {
+  Ds1ToDstarK0s = 0,
+  Ds2StarToDplusK0s,
+  XcToDplusLambda,
+  LambdaDminus
+};
+
+struct HfTaskReso {
+  Configurable<float> ptMinReso{"ptMinReso", 5, "Discard events with smaller pT"};
+  Configurable<bool> cutBeforeMixing{"cutBeforeMixing", false, "Apply pT cut to candidates before event mixing"};
+  Configurable<bool> cutAfterMixing{"cutAfterMixing", false, "Apply pT cut to candidates after event mixing"};
+  // Configurables axis for histos
+  ConfigurableAxis configAxisPt{"configAxisPt", {VARIABLE_WIDTH, 0., 1.f, 2.f, 3.f, 4.f, 5.f, 6.f, 8.f, 12.f, 24.f, 50.f}, "axis for pT"};
+  ConfigurableAxis configAxisPtProng0{"configAxisPtProng0", {VARIABLE_WIDTH, 0., 1.f, 2.f, 3.f, 4.f, 5.f, 6.f, 8.f, 12.f, 24.f, 50.f}, "axis for pT of D daughter"};
+  ConfigurableAxis configAxisPtProng1{"configAxisPtProng1", {VARIABLE_WIDTH, 0., 1.f, 2.f, 3.f, 4.f, 5.f, 6.f, 8.f, 12.f, 24.f, 50.f}, "axis for pT of V0 daughter"};
+  ConfigurableAxis configAxisInvMassReso{"configAxisInvMassReso", {200, 2.34, 2.74}, "axis for Resonance invariant mass"};
+  ConfigurableAxis configAxisInvMassProng0{"configAxisInvMassProng0", {175, 1.70, 2.05}, "axis for Prong0 invariant mass"};
+  ConfigurableAxis configAxisInvMassProng1{"configAxisInvMassProng1", {80, 0.46, 0.54}, "axis for Prong1 invariant mass"};
+  ConfigurableAxis configAxisCosThetaStar{"configAxisCosThetaStar", {40, -1, 1}, "axis for V0 costheta star"};
+  ConfigurableAxis configAxisBkgBdtScore{"configAxisBkgBdtScore", {100, 0, 1}, "axis for D meson Bkg BDT score"};
+  ConfigurableAxis configAxisNonPromptBdtScore{"configAxisNonPromptBdtScore", {100, 0, 1}, "axis for D meson Non Prompt BDT score"};
+  // Configurables for ME
+  Configurable<int> numberEventsMixed{"numberEventsMixed", 5, "Number of events mixed in ME process"};
+  Configurable<int> numberEventsToSkip{"numberEventsToSkip", -1, "Number of events to Skip in ME process"};
+  ConfigurableAxis multPoolBins{"multPoolBins", {VARIABLE_WIDTH, 0., 45., 60., 75., 95, 250}, "event multiplicity pools (PV contributors for now)"};
+  ConfigurableAxis zPoolBins{"zPoolBins", {VARIABLE_WIDTH, -10.0, -4, -1, 1, 4, 10.0}, "z vertex position pools"};
+  // ConfigurableAxis bzPoolBins{"bzPoolBins", {2, -10, 10}, "Bz of collision"};
+
+  using reducedResoWithMl = soa::Join<aod::HfCandCharmReso, aod::HfCharmResoMLs>;
+  SliceCache cache;
+  Preslice<reducedResoWithMl> resoPerCollision = aod::hf_track_index_reduced::hfRedCollisionId;
+
+  // Histogram Registry
+  HistogramRegistry registry;
+
+  // init
+  void init(InitContext&)
+  {
+    const AxisSpec axisPt{configAxisPt, "#it{p}_{T} (GeV/#it{c})"};
+    const AxisSpec axisPtProng0{configAxisPtProng0, "#it{p}_{T} (GeV/#it{c})"};
+    const AxisSpec axisPtProng1{configAxisPtProng1, "#it{p}_{T} (GeV/#it{c})"};
```

Comment:
Why is this conversion needed?

Reply:
I believe it's main purpose it is to change the label of the axis wrt. the configurable axis description displayed on HL. It seems common practice in several HF tasks.

Reply:
Thanks for the clarification, I must have missed this practice. If it works with setting the title in `ConfigurableAxis` and using it directly in the histogram declaration, I would go for it.